### PR TITLE
Fix hardhat artifacts collection

### PR DESCRIPTION
**Pull request by Nurchik**
* Ignore library contracts by default

### CODE REVIEW

File: fuzzing_cli/fuzz/ide/hardhat.py [hunk 1]
```diff
@@ -24,3 +31,132 @@ def get_default_build_dir() -> Path:
...
+
+    @lru_cache(maxsize=1)
+    def process_artifacts(self) -> Tuple[Dict[str, List[Contract]], Dict[str, Source]]:
+        result_contracts = defaultdict(list)
+        result_sources = {}
+
+        build_info_paths: Dict[Path, Dict[str, List[str]]] = defaultdict(
+            partial(defaultdict, list)
+        )
+        for source_file_name, contracts in self.build_info.items():
```
Comment (cd1m0) [inline fuzzing_cli/fuzz/ide/hardhat.py:96]: Can we add code here that:

1. Detects if the same `source_file_name` appears in multiple build_info files
2. Communicates a warning/error to the FaaS API when this happens

I want to be able to detect when this happens in practice so we can debug it and adapt the code

**Review by cd1m0 — CHANGES_REQUESTED** [review]
Good work! Some small nits requested.

I think this will work as a patch over. But eventually we need to re-think the information we send to the FaaS API.

In a perfect world, I just want to send a list of build artifacts `B`, and a mapping `M` from deployed contract addresses to a tuple `[idx, srcFile, contractName]`. Then for each deployed contract, the ast information is in `B[idx][srcFile][contractName]["ast"]`. 

In the FaaS backend, we can create a separate solc-typed-ast for each entry `B[i]`, and we can be sure that the `id`s are unique. 

However this is a longer discussion, that needs to involve more people (@wuestholz, @JoranHonig).

File: fuzzing_cli/fuzz/ide/generic.py [hunk 2]
```diff
@@ -196,7 +199,41 @@ def get_contract(self, deployed_bytecode: str) -> Optional[Contract]:
...
+                if node["nodeType"] != "ContractDefinition":
+                    continue
+                contract_name = node["name"]
+                types_mapping[source_file_name][contract_name] = node["contractKind"]
+        return types_mapping
+
+    def _get_contract_kind(self, contract: Contract) -> ContractKind:
+        return (
+            self._contracts_kind_mapping().get(contract["mainSourceFile"], {})
+            # if contract is not found in the mapping, we assume it's a contract
```
[the remark this answers is not in the document]
Comment (Nurchik) [reply fuzzing_cli/fuzz/ide/generic.py:227]: it shouldn't happen, but it's an additional safe measure

**Review by cd1m0 — APPROVED** [review]
lgtm

**Pull request merged**